### PR TITLE
update:cookies-headers-logic

### PR DESCRIPTION
**Pull request by admirsaheta**
### WHY are these changes introduced?

Extracted the logic for checking if the cookie and signature cookie exist into a separate function for better readability.

Used const instead of let wherever possible to make code more predictable and prevent accidental reassignment.

### WHAT is this pull request doing?

Cleaner readability and faster execution time on helper functions.

## Type of change

- [x] Patch: Bug (non-breaking change which fixes an issue)
- [ ] Minor: New feature (non-breaking change which adds functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [ ] I have added/updated tests for this change
- [ ] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

File: packages/apps/shopify-api/runtime/http/cookies.ts [hunk 1]
```diff
@@ -175,25 +175,40 @@ export class Cookies {
...
-      this.deleteCookie(signedCookieName);
-      this.deleteCookie(cookieName);
+    if (
+      !this.cookieExists(cookieName) ||
+      !this.cookieExists(signedCookieName)
+    ) {
+      this.deleteInvalidCookies(cookieName, signedCookieName);
+      return false;
+    }
+    const cookieValue = this.get(cookieName);
```
Comment (lizkenyon) [inline packages/apps/shopify-api/runtime/http/cookies.ts:195]: Are lines 195-199 duplicating lines 188-194?
I am not sure I am understanding why both are necessary?

**Review by lizkenyon — COMMENTED** [review]
Could you please also check on the linting issues?

If you would be interested in solving some issue other than refactoring let us know and we can tag some good first issues.

**Comment by admirsaheta** [comment]
Of course, feel free to tag me on some 👍  
Ran prettier, all is fine now. @lizkenyon 

As for the duplicate code question, ensuring both existence and content validity checks mitigates potential risks associated with cookie manipulation or unexpected data states. It enhances the function's resilience by addressing not just absence but also quality of data, thereby improving overall reliability in diverse deployment environments.

File: packages/apps/shopify-api/runtime/http/cookies.ts [hunk 2]
```diff
@@ -177,23 +177,38 @@ export class Cookies {
 
   async isSignedCookieValid(cookieName: string): Promise<boolean> {
     const signedCookieName = `${cookieName}.sig`;
-    // No cookie or no signature cookie makes the cookie it invalid.
-    if (!this.get(cookieName) || !this.get(signedCookieName)) {
-      this.deleteCookie(signedCookieName);
-      this.deleteCookie(cookieName);
+    if (
+      !this.cookieExists(cookieName) ||
```
Comment (lizkenyon) [inline packages/apps/shopify-api/runtime/http/cookies.ts:181]: This line evaluates to `Boolean(this.get(cookieName)`, which evaluates to `Boolean(this.receivedCookieJar[name]?.value)`.

You then are checking right after for truthy value
`const cookieValue = this.get(cookieName);` which is equivalent to this `this.receivedCookieJar[name]?.value`

If you think that both these checks are necessary could you explain the scenario, or even better write a test case to demonstrate the scenario.

Comment (admirsaheta) [reply packages/apps/shopify-api/runtime/http/cookies.ts:181]: Added test cases 👍

File: packages/apps/shopify-api/runtime/http/cookies.ts [hunk 3]
```diff
@@ -177,23 +177,38 @@ export class Cookies {
...
+      !this.cookieExists(signedCookieName)
+    ) {
+      this.deleteInvalidCookies(cookieName, signedCookieName);
+      return false;
+    }
+    const cookieValue = this.get(cookieName);
+    const signature = this.get(signedCookieName);
+
+    if (!cookieValue || !signature) {
```
Comment (lizkenyon) [inline packages/apps/shopify-api/runtime/http/cookies.ts:190]: I see now that this is necessary to satisfy typescript.

**Review by lizkenyon — APPROVED** [review]

**Pull request merged**